SonarQube.Client/Api/V7_20/GetIssuesRequest.cs

```diff
@@ -47,19 +47,26 @@ protected override SonarQubeIssue[] ParseResponse(string response)
         {
             var root = JObject.Parse(response);
 
-            // Lookup component key -> path for files. Each response contains normalized data, containing
-            // issues and components, where each issue's "component" property points to a component with
-            // the same "key". We obtain the FilePath of each issue from its corresponding component.
-            var componentKeyPathLookup = GetComponentKeyPathLookup(root);
+            // This is a paged request so ParseResponse will be called once for each "page"
+            // of the response. However, we expect each page to be self-contained, so we want
+            // to rebuild the lookup each time.
+            componentKeyPathLookup = GetComponentKeyPathLookup(root);
 
             return root["issues"]
                 .ToObject<ServerIssue[]>()
-                .Select(issue => ToSonarQubeIssue(issue, componentKeyPathLookup))
+                .Select(ToSonarQubeIssue)
                 .ToArray();
         }
 
         #region Json data classes -> public read-only class conversion methods
 
+        /// <summary>
+        /// Lookup component key -> path for files. Each response contains normalized data, containing
+        /// issues and components, where each issue's "component" property points to a component with
+        /// the same "key". We obtain the FilePath of each issue from its corresponding component.
+        /// </summary>
+        private ILookup<string, string> componentKeyPathLookup;
```

Comment:
I made the lookup an instance variable since it is now needed in more places, and passing it through a chain of `static` methods was getting ugly and hard to read.